rpm/update-spec-version.sh

```diff
@@ -4,12 +4,19 @@
 # default. Useful for local manual rpm builds where the Version needs to be set
 # correctly.
 
-set -e
+set -eox pipefail
 
-# Script is run from git root directory
-SPEC_FILE=rpm/podman.spec
+PACKAGE=podman
+SPEC_FILE=$PACKAGE.spec
+GIT_DESCRIBE=$(git describe)
+VERSION=$(echo $GIT_DESCRIBE | sed -e 's/^v//' -e 's/-/~/g')
 
-LATEST_TAG=$(git tag --sort=creatordate | tail -1)
-LATEST_VERSION=$(echo $LATEST_TAG | sed -e 's/^v//')
+# Update spec file to use local changes
+sed -i "s/^Version:.*/Version: $VERSION/" $SPEC_FILE
+sed -i "s/^Source0:.*/Source0: $PACKAGE-$GIT_DESCRIBE.tar.gz/" $SPEC_FILE
+sed -i "s/^%autosetup.*/%autosetup -Sgit -n %{name}-$GIT_DESCRIBE/" $SPEC_FILE
 
-sed -i "s/^Version:.*/Version: $LATEST_VERSION/" $SPEC_FILE
+# Generate Source0 archive from HEAD
+cd ..
+git archive --prefix=$PACKAGE-$GIT_DESCRIBE/ HEAD -o rpm/$PACKAGE-$GIT_DESCRIBE.tar.gz
+cd rpm
```

Comment:
That final line is a NOP, causing the reader to waste time wondering why it's there. Should you need to repush for another reason (this one isn't a blocker), I would recommend
```
(cd .. && git archive ....)
```
(open paren, `cd ..`, same archive command, close paren). `git -C` would work too but then I think the `rpm/PATH` is less clear.

Reply:
done. I also added a `--format=tar.gz` . PTAL